### PR TITLE
[Connector-V2][JDBC] Support database: greenplum

### PR DESCRIPTION
**Pull request by hailin0**
Support connect greenplum drivers:
* postgresql driver: org.postgresql.Driver
* greenplum driver: com.pivotal.jdbc.GreenplumDriver

## Purpose of this pull request

Support jdbc connect greenplum  #1946

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/jdbc/FakeSourceToJdbcGreenplumIT.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+@SuppressWarnings("checkstyle:MagicNumber")
```
Comment (ashulin) [inline seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/jdbc/FakeSourceToJdbcGreenplumIT.java]:
```suggestion
```
Currently IT and Test classes will automatically ignore magic number. 
If it cannot be automatically ignored, please give me feedback.

Comment (hailin0) [reply seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/jdbc/FakeSourceToJdbcGreenplumIT.java]: 👌 ok

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/greenplum/GreenplumDialectFactory.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+import com.google.auto.service.AutoService;
+
+@AutoService(JdbcDialectFactory.class)
+public class GreenplumDialectFactory implements JdbcDialectFactory {
+
+    @Override
+    public boolean acceptsURL(String url) {
+        // Support greenplum native driver: com.pivotal.jdbc.GreenplumDriver
+        return url.startsWith("jdbc:pivotal:greenplum:");
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/greenplum/GreenplumDialectFactory.java:33]: If `url` must not be `null`, please use `@Nonnull` to mark it.

**Comment by EricJoy2048** [comment]
Please add the `document` of this connector.

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/greenplum/GreenplumDialectFactory.java (hunk 2 above)
Comment (hailin0) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/greenplum/GreenplumDialectFactory.java:33]: ok

**Comment by hailin0** [comment]
> Please add the `document` of this connector.

updated

File: docs/en/connector-v2/sink/Greenplum.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
+# Greenplum
+
+> Greenplum sink connector
```
Comment (ic4y) [inline docs/en/connector-v2/sink/Greenplum.md:3]: Need to indicate whether two-phase commit is supported, which is related to exactly-once semantics.
How to configure `xa_data_source_class_name` if you perform a two-phase commit.
It's better to add a job configuration example.

File: seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/jdbc/JdbcGreenplumIT.java [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+        List<Object> result = new ArrayList<>();
+        try (Statement statement = jdbcConnection.createStatement()) {
+            ResultSet resultSet = statement.executeQuery(sql);
+            while (resultSet.next()) {
+                result.add(Arrays.asList(
+                        resultSet.getInt("age"),
+                        resultSet.getString("name")));
+            }
+        }
+        Assertions.assertEquals(100, result.size());
```
Comment (ic4y) [inline seatunnel-e2e/seatunnel-flink-connector-v2-e2e/src/test/java/org/apache/seatunnel/e2e/flink/v2/jdbc/JdbcGreenplumIT.java]: Only check the amount of data ? Is it possible to add data type and value validation

**Review by ic4y — COMMENTED** [review]
Can we also add tests for Stream mode and tests for XA transaction(two-phase commit)

File: docs/en/connector-v2/sink/Greenplum.md (hunk 3 above)
Comment (hailin0) [reply docs/en/connector-v2/sink/Greenplum.md:3]: XA transaction is not yet supported in Greenplum Database

Comment (hailin0) [reply docs/en/connector-v2/sink/Greenplum.md:3]: https://docs.vmware.com/en/VMware-Tanzu-Greenplum/6/greenplum-database/GUID-ref_guide-feature_summary.html

**Comment by hailin0** [comment]
> Can we also add tests for Stream mode and tests for XA transaction(two-phase commit)

@ic4y updated

**Comment by ic4y** [comment]
@hailin0 JDBC Sink supports Stream mode. Can it be covered in tests？

**Comment by hailin0** [comment]
> @hailin0 JDBC Sink supports Stream mode. Can it be covered in tests？

Stream mode only dedicated to source?

**Comment by ic4y** [comment]
> > @hailin0 JDBC Sink supports Stream mode. Can it be covered in tests？
> 
> Stream mode only dedicated to source?

No. Stream mode only dedicated to Sink

**Comment by hailin0** [comment]
> > > @hailin0 JDBC Sink supports Stream mode. Can it be covered in tests？
> > 
> > 
> > Stream mode only dedicated to source?
> 
> No. Stream mode only dedicated to Sink

Not support exactly-once semantics (XA transaction is not yet supported in Greenplum database).

**Comment by hailin0** [comment]
@CalvinKirs please review

**Comment by ic4y** [comment]
@hailin0 Thanks
+1

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**